### PR TITLE
[Clover step-16] Add users

### PR DESCRIPTION
**Pull request by clover-zhao**
### Summary/概要
Description:
Implemented user logic include user controller and sign up view
Changes Made:
- create user controller and views
- create signup page and logic
- apply style to signup page
- add user creation code in seed file
- add `bullet` gem and config
- add `bcrypt` gem for using active model `has_secure_password`

### How to check the behavior/動作確認
Sign up page
image

### CODE REVIEW

**Comment by wataru0** [comment]
nice sign-up page! 👍

File: myapp/app/models/user.rb [hunk 1]
```diff
@@ -2,9 +2,10 @@ class User < ApplicationRecord
     ROLES = ["user", "admin"].freeze
 
     validates :username, presence: true
-    validates :email, presence: true
-    validates :password_hash, presence: true
+    validates :email, presence: true, uniqueness: true
```
Comment (ryo-ninomiya) [inline myapp/app/models/user.rb:5]: [nits]
I recommend that the scheme of users' email columns also be changed to uniq index.

**Review by ryo-ninomiya — APPROVED** [review]
I commented one thing, but it's not a big deal, so I approve it!
image

File: myapp/app/models/user.rb (hunk 1 above)
Comment (clover-zhao) [reply myapp/app/models/user.rb:5]: Sure! Thanks for your advice.

**Pull request merged**